domain-models-runtime/src/main/java/org/folio/rest/RestVerticle.java

```diff
@@ -155,338 +62,78 @@ public void start(Promise<Void> startPromise) throws Exception {
-      // URL if no match is found, then the requested url is not supported by
-      // the ramls and we return an error - this has positive security implications as well
-      for (String urlPath : urlPaths) {
-        String[] pathParams = matchPath(rc.request().path(), regex2Pattern.get(urlPath));
-        if (pathParams != null) {  // if matched
-          validPath = true;
-
-          // get the function that should be invoked for the requested
-          // path + requested http_method pair
-          JsonObject ret = mappedURLs.getMethodbyPath(urlPath, rc.request().method().toString());
-          // if a valid path was requested but no function was found
-          if (ret == null) {
-
-            // if the path is valid and the http method is options
-            // assume a cors request
-            if (rc.request().method() == HttpMethod.OPTIONS) {
-              rc.response().end();
-              return;
-            }
-
-            // the url exists but the http method requested does not match a function
-            // meaning url+http method != a function
-            endRequestWithError(rc, 400, true, messages.getMessage("en", MessageConsts.HTTPMethodNotSupported),
-              validRequest);
-            return;
+    //single handler for all url calls other then documentation
+    //which is handled separately
+    //router.routeWithRegex("^(?!.*apidocs).*$").handler(rc -> route(mappedURLs, urlPaths, regex2Pattern, rc));
+    // routes requests on “/assets/*” to resources stored in the “assets”
+    // directory.
+    router.route("/assets/*").handler(StaticHandler.create("assets"));
+
+    // In the following example all requests to paths starting with
+    // /apidocs/ will get served from the directory resources/apidocs:
+    // example:
+    // http://localhost:8181/apidocs/index.html?raml=raml/_patrons.raml
+    router.route("/apidocs/*").handler(StaticHandler.create("apidocs"));
+    // startup http server on port 8181 to serve documentation
+
+    //if client includes an Accept-Encoding header which includes
+    //the supported compressions - deflate or gzip.
+    HttpServerOptions serverOptions = new HttpServerOptions();
+    serverOptions.setCompressionSupported(true);
+
+    HttpServer server = vertx.createHttpServer(serverOptions);
+    String portS = System.getProperty(HTTP_PORT_SETTING);
+    int port;
+    if (portS != null) {
+      port = Integer.parseInt(portS);
+      config().put(HTTP_PORT_SETTING, port);
+    } else {
+      // we are here if port was not passed via cmd line
+      port = config().getInteger(HTTP_PORT_SETTING, 8081);
+    }
+    RestRouting.populateRoutes(router, packageOfImplementations)
+        .compose(x -> runHook())
+        .compose(x -> {
+          log.info("Listening port {}", port);
+          return server.requestHandler(router).listen(port);
+        })
+        .compose(ret -> {
+          try {
```

Comment:
The 2 try-catch clauses within the .compose are not needed because .compose catches all exceptions.
This is possible if runPeriodicHook() and runPostDeployHook(Handler) no longer throw (checked) Exception - simply add
```
    } catch (IOException|IllegalArgumentException|ReflectiveOperationException e) {
      throw new RuntimeException(e);
     }
```
```
    } catch (IOException|ReflectiveOperationException e) {
      throw new RuntimeException(e);
    }
```
to them to throw unchecked exceptions only.